src/core/Dependency.ml

```diff
@@ -0,0 +1,40 @@
+(* The "core of a version": a dot separated list of numbers, like 4.1.6.2.7 *)
+type version_core = { major : int; minor : int; incrementals : int list }
+
+(* Versions are sometimes listed as arbitrary strings, like a github URL *)
+and version = Version of version_core | Other of string
+and constraint_ = Eq | Gte | Lte | Gt | Lt
+
+(* Something like (>= 2.0.0) or (== 5.1.7) *)
+and version_constraint = { version : version; constraint_ : constraint_ }
+
+(* An intersection of constraints, like (>= 1.0.0, < 3.0.0), meaning "greater than or equal 1.0.0 and less than 3.0.0" *)
+(* We don't have union/an actual tree of constraints because of the historical baggage of the structure of supply chain rules,
+   which only have top-level union.
+*)
+and constraint_ast = And of version_constraint list
+
+and t = {
+  package_name : string;
+  package_version : version;
+  package_version_string : string;
+  ecosystem : ecosystem;
+  transitivity : transitivity;
+  url : string option;
```

Comment:
Uri.t?

Reply:
+1. Also, it's not immediately clear to me what this url would be for. Is it the location in the registry where the files for the dependency could be obtained? A comment or more descriptive name would be nice.